lib/motion/revision_calculator.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "digest"
+require "motion"
+
+module Motion
+  class RevisionCalculator
+    attr_reader :revision_paths
+
+    def initialize(revision_paths:)
+      @revision_paths = revision_paths
+    end
+
+    def perform
+      derive_md5_hash
+    end
+
+    private
+
+    def file_contents
+      files.map { |f| File.read(f) }.join
```

Comment:
This is going to load all of the files into memory at once, but I think it should be possible to stream them into the digest one by one:

```ruby
digestor = Digest::SHA256.new

Dir["**/*"].each do |file|
  digestor << File.read(file) if File.file?(file)
end

digestor.hexdigest
```